src/main/rx/nrf24_syma.c

```diff
@@ -0,0 +1,293 @@
+        ret = -ret;
+    }
+    return (uint16_t)(PWM_RANGE_MIDDLE + ret);
+}
+
+STATIC_UNIT_TESTED bool checkBindPacket(const uint8_t *packet)
+{
+    bool bindPacket = false;
+    if (symaProtocol == NRF24RX_SYMA_X) {
+        if ((packet[5] == 0xaa) && (packet[6] == 0xaa) && (packet[7] == 0xaa)) {
+            bindPacket = true;
+            rxTxAddr[4] = packet[0];
+            rxTxAddr[3] = packet[1];
+            rxTxAddr[2] = packet[2];
+            rxTxAddr[1] = packet[3];
+            rxTxAddr[0] = packet[4];
+        }
+    } else {
+        if ((packet[0] == 0) && (packet[1] == 0) && (packet[14] == 0xc0) && (packet[15] == 0x17)) {
+            bindPacket = true;
+        }
+    }
+    return bindPacket;
+}
+
+void symaSetRcDataFromPayload(uint16_t *rcData, const uint8_t *packet)
+{
+    for (int ii = NRF24_AUX1; ii <= NRF24_AUX5; ++ii) {
+        rcData[ii] =  PWM_RANGE_MIN;
+    }
+    rcData[NRF24_THROTTLE] = convertToPwmUnsigned(packet[0]);
+    rcData[NRF24_ROLL] = convertToPwmSigned(packet[3]); // aileron
+    if (symaProtocol == NRF24RX_SYMA_X) {
+        rcData[NRF24_PITCH] = convertToPwmSigned(packet[1]); // elevator
+        rcData[NRF24_YAW] = convertToPwmSigned(packet[2]); // rudder
+        if (packet[4] & 0x40) { // picture
+            rcData[NRF24_AUX4] = PWM_RANGE_MAX;
+        }
+        if (packet[4] & 0x80) { // video
+            rcData[NRF24_AUX5] = PWM_RANGE_MAX;
+        }
+        if (packet[5] & 0xc0) { // rate
+            rcData[NRF24_AUX3] = PWM_RANGE_MAX;
+        }
+        if (packet[6] & 0x40) { // flip
+            rcData[NRF24_AUX2] = PWM_RANGE_MAX;
+        }
+        if (packet[7] & 0x80) { // headless
+            rcData[NRF24_AUX1] = PWM_RANGE_MAX;
+        }
+    } else {
+        rcData[NRF24_PITCH] = convertToPwmSigned(packet[2]) ^ 0x80; // elevator
+        rcData[NRF24_YAW] = convertToPwmSigned(packet[1]); // rudder
+        if (packet[14] & 0x08) { // picture
+            rcData[NRF24_AUX4] = PWM_RANGE_MAX;
+        }
+        if (packet[14] & 0x10) { // video
+            rcData[NRF24_AUX5] = PWM_RANGE_MAX;
+        }
+        if (packet[14] & 0x01) { // flip
+            rcData[NRF24_AUX2] = PWM_RANGE_MAX;
+        }
+        if (packet[14] & 0x04) { // rate
+            rcData[NRF24_AUX3] = PWM_RANGE_MAX;
+        }
+    }
+}
+
+static void hopToNextChannel(void)
+{
+    // hop channel every second packet
+    ++packetCount;
+    if ((packetCount & 0x01) == 0) {
+        ++rfChannelIndex;
+        if (rfChannelIndex >= rfChannelCount) {
+            rfChannelIndex = 0;
+        }
+    }
+    NRF24L01_SetChannel(rfChannels[rfChannelIndex]);
+}
+
+// The SymaX hopping channels are determined by the low bits of rxTxAddress
+void setSymaXHoppingChannels(uint32_t addr)
+{
+    addr = addr & 0x1f;
+    if (addr == 0x06) {
+        addr = 0x07;
+    }
+    const uint32_t inc = (addr << 24) | (addr << 16) | (addr << 8) | addr;
+    uint32_t * const prfChannels = (uint32_t *)rfChannels;
```

Comment:
`rfChannels` may be unaligned, causing hard to find problems. Better use uint32_t temporary and memcpy it to `rfChannels`